calico/_includes/charts/calico/templates/calico-node-rbac.yaml

```diff
@@ -70,6 +70,14 @@ rules:
       - pods/status
     verbs:
       - patch
+  # Used for creating service account tokens to be used by the CNI plugin
+  - apiGroups: [""]
+    resources:
+      - serviceaccounts/token
+    resourceNames:
+      - calico-node
```

Comment:
If the role containing this rule is also bound to the `calico-node` service account, it allows `calico-node` to mint tokens for `calico-node`. This creates a potential security issue where an attacker could steal a token and prevent their access from ever expiring by sending periodic TokenRequests for the same service account whose token they stole. Instead, it would be better to grant `calico-node` permission to request tokens for a separate service account that the CNI is intended to run as. For example, grant `calico-node` permission to mint tokens for a `calico-node-cni` service account, and grant `calico-node-cni` the permissions that the CNI needs. That way, no tokens have the ability to self-perpetuate.

Reply:
Yep, I agree. I hadn't spotted this particular issue but already have a ticket tracking this enhancement for other reasons as well: https://github.com/projectcalico/calico/issues/5921